runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -508,44 +508,97 @@ uint32_t TR::ARM64VirtualUnresolvedSnippet::getLength(int32_t estimatedSnippetSt
 
 uint8_t *TR::ARM64InterfaceCallSnippet::emitSnippetBody()
    {
+   TR::Compilation *comp = cg()->comp();
    uint8_t *cursor = cg()->getBinaryBufferCursor();
+   TR::Node *callNode = getNode();
    TR::SymbolReference *methodSymRef = getNode()->getSymbolReference();
    TR::SymbolReference *glueRef = cg()->symRefTab()->findOrCreateRuntimeHelper(TR_ARM64interfaceCallHelper, false, false, false);
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(comp->fe());
+   void* thunk = fej9->getJ2IThunk(callNode->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod(), comp);
 
    getSnippetLabel()->setCodeLocation(cursor);
 
    // bl glueRef
-   *(int32_t *)cursor = cg()->encodeHelperBranchAndLink(glueRef, cursor, getNode());
-   cursor += 4;
+   *(int32_t *)cursor = cg()->encodeHelperBranchAndLink(glueRef, cursor, callNode);
+   uint8_t *blAddress = cursor;
+   cursor += ARM64_INSTRUCTION_LENGTH;
 
    // Store the code cache RA
    *(intptrj_t *)cursor = (intptrj_t)getReturnLabel()->getCodeLocation();
    cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(
                                cursor,
                                NULL,
                                TR_AbsoluteMethodAddress, cg()),
-                               __FILE__, __LINE__, getNode());
-   cursor += 8;
+                               __FILE__, __LINE__, callNode);
+   cursor += sizeof(intptrj_t);
 
    // CP
-   *(intptrj_t *)cursor = (intptrj_t)methodSymRef->getOwningMethod(cg()->comp())->constantPool();
-   cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(
-                               cursor,
-                               *(uint8_t **)cursor,
-                               getNode() ? (uint8_t *)getNode()->getInlinedSiteIndex() : (uint8_t *)-1,
-                               TR_Thunks, cg()),
-                               __FILE__, __LINE__, getNode());
-   cursor += 8;
+   intptrj_t cpAddr = (intptrj_t)methodSymRef->getOwningMethod(comp)->constantPool();
+   *(intptrj_t *)cursor = cpAddr;
+   uint8_t *j2iThunkRelocationPoint = cursor;
+   cursor += sizeof(intptrj_t);
 
    // CP index
    *(intptrj_t *)cursor = methodSymRef->getCPIndexForVM();
-   cursor += 8;
+   cursor += sizeof(intptrj_t);
 
-   // Add 2 more slots for resolved values (interface class and iTable offset)
+   // 2 slots for resolved values (interface class and iTable index)
    *(intptrj_t *)cursor = 0;
-   cursor += 8;
+   cursor += sizeof(intptrj_t);
    *(intptrj_t *)cursor = 0;
-   cursor += 8;
+   cursor += sizeof(intptrj_t);
+
+   // Initialize for: two class ptrs, two target addrs
+   // Initialize target addrs with the address of the bl
+   *(intptrj_t *)cursor = -1;
+   *(intptrj_t *)(cursor+sizeof(intptrj_t)) = (intptrj_t)blAddress;
+   *(intptrj_t *)(cursor+2*sizeof(intptrj_t)) = -1;
+   *(intptrj_t *)(cursor+3*sizeof(intptrj_t)) = (intptrj_t)blAddress;
```

Comment:
What is the purpose of these fields?  Are they for building an interface PIC?  If I understand the code in this PR correctly, they are introduced but not used in this PR.  I would say your commit message should at least comment on the fact that you're introducing them here, but the actual PIC mechanics will be completed later.

Reply:
Updated the code removing the unused snippet fields.